src/vmod_saintmode.c

```diff
@@ -118,9 +118,9 @@ vmod_denylist(VRT_CTX, struct vmod_priv *priv, VCL_DURATION expires) {
 	}
 
 	CHECK_OBJ_NOTNULL(ctx, VRT_CTX_MAGIC);
-	if (ctx->method != VCL_MET_BACKEND_RESPONSE) {
+	if (!ctx->bo || !ctx->bo->director_resp) {
```

Comment:
actually, we can get rid of this, and just use `$Restrict` in the VCC.
@delthas, do you want to do it, or should I?